### DISCUSSION TITLE
Contributing a parser to the new "Bluetooth Devices" org

### DISCUSSION THREAD

**Discussion by apt-itude** [custom-components/ble_monitor#990] [Q&A]
Hi 👋 
I have implemented the Tilt parser using the new Bluetooth integration, following the format of other parser repos [here](https://github.com/Bluetooth-Devices). I'd like to contribute it if possible. How can I create a new repo in that org?

**Comment by Ernst79** [comment]
Excellent. I can add you as a developer, will do that later today.

**Comment by Ernst79** [comment]
You should have an invite in your mail.

**Comment by claudiopolis** [comment]
I'd like to develop and contribute a parser for a home treadmill I just got. Is there a guide somewhere for how to do the BLE sniffing/capture and other reverse engineering necessary to achieve this?

**Reply by Ernst79** [reply]
For the data sniffing, you can use this page https://custom-components.github.io/ble_monitor/sensor_request It explains some possibilities to get the data. Look for the manufacturer specific data (FF) or the service data (16) parts, these contain the data. Make a long log, try to understand which bytes changes. If you post the data in an new issue, I can help you with that

After that, the steps are
Add a parser to the parser folder (copy one of the others)
Add it to the init.py file in the parser folder, where we filter the incoming data.
If the parser works, last step is to add the sensor to const.py in the main folder.

**Reply by claudiopolis** [reply]
Thanks very much @Ernst79 !
